### PR TITLE
ENH: Increase `itk::MaskFeaturePointSelectionFilter` class coverage

### PR DESCRIPTION
**Pull request by jhlegarreta**
Increase `itk::MaskFeaturePointSelectionFilter` class coverage: provide
a mask image to check the filter behavior when a mask is provided, and
exercise the corresponding `Get` method.

Add the accompanying mask input and baseline images.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [X] Added test (or behavior not changed)

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Not sure why the CIs are reporting a failure. The test is passing locally on my Win 7, MSVC 2019 machine

itkMaskFeaturePointSelectionFilterTestMask

**Comment by jhlegarreta** [comment]
> No, it is not. I think that all file formats store just one region (buffered one), and set the origin of the image to the start index of that region. And when reading, they make the saved image be the largest, and index is implicitly zero.

:roll_eyes: OK, so maybe I got this wrong and the previous implementation was OK, and the mask image is always assumed to have the size of the input image, with the region of interest/binary region box placed at the desired particular location. I'll see during the weekend if I can do this.

File: Modules/Filtering/ImageFeature/include/itkMaskFeaturePointSelectionFilter.hxx [hunk 1]
```diff
@@ -146,7 +146,7 @@ MaskFeaturePointSelectionFilter<TImage, TMask, TFeatures>::GenerateData()
   else
   {
     // copy mask into selectionMap
-    ImageRegionConstIterator<MaskType>    maskItr(mask, region);
+    ImageRegionConstIterator<ImageType>   maskItr(image, mask->GetLargestPossibleRegion());
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply Modules/Filtering/ImageFeature/include/itkMaskFeaturePointSelectionFilter.hxx:149]: I'd assume the mask is a binary image, and only serves to located the voxels that need to be taken into account for the computation. But as said in [the comment below](https://github.com/InsightSoftwareConsortium/ITK/pull/3349#issuecomment-1084766978), maybe I was wrong in my mask assumptions/understanding.

**Comment by dzenanz** [comment]
I don't know anything about this filter, so I don't know why [non-connectivity of value 3](https://open.cdash.org/test/656592635) exception occurs at https://github.com/InsightSoftwareConsortium/ITK/blob/v5.3rc03/Modules/Filtering/ImageFeature/include/itkMaskFeaturePointSelectionFilter.hxx#L100-L101.

File: Modules/Filtering/ImageFeature/test/Input/HeadMRVolumeMaskImage.mha.sha512 [hunk 2]
```diff
@@ -0,0 +1 @@
+1f24f6f43de351147be839bb4dd8dbd5c8868b27b4438f9653c835296c23aff9afe39ec6b4946f81cc6e41c74017041cfb385440007c6910c686a70dafeeb9ae
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply Modules/Filtering/ImageFeature/test/Input/HeadMRVolumeMaskImage.mha.sha512]: Saved the new mask as a compressed file. Thanks.

File: Modules/Filtering/ImageFeature/include/itkMaskFeaturePointSelectionFilter.hxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply Modules/Filtering/ImageFeature/include/itkMaskFeaturePointSelectionFilter.hxx]: So following the mentioned comment, it turns out it was not a bug indeed 😊: it was just my limited knowledge/ignorance.

**Comment by jhlegarreta** [comment]
> I don't know anything about this filter, so I don't know why [non-connectivity of value 3](https://open.cdash.org/test/656592635) exception occurs

My knowledge of the filter is also limited; I have not investigated about the connectivity aspects; I assume there is some reason for that. So "Carry on and increase coverage" 😄 .

Thanks for the review @dzenanz. Greatly appreciated :gift:.

**Review by thewtex — APPROVED** [review]
:green_heart: :diving_mask:

**Pull request merged**